### PR TITLE
Api feature

### PR DESCRIPTION
**Pull request by Misori-simon**
### This PR contains the following changes
- Project Setup (linters, webpack)
- Add getCityWeather function to talk to api 
- Add createForm and displayResult functions to update the dom and trigger API communication
- Add addUnit, displayErrorMessage, clearForm functions
- Update Readme

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import getCityWeather from './weather_api';
+import createForm from './dom';
+
+const appId = 'f2526aabeeb777ad3fbb777842b7c51e';
+
+createForm(getCityWeather, appId);
+
+// console.log(getCityWeather('london', 'metric'));
```
Comment (uryelah) [inline src/index.js]: - [x] Lets keep the code base clean and organized, remove the commented out code.

File: src/weather_api.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  results.cloudState = json.weather[0].main;
+  results.cloudStateDesc = json.weather[0].description;
+  results.cloudStateIcon = json.weather[0].icon;
+
+  return results;
+};
+
+const getCityWeather = async (city, unit, appId) => {
+  const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&units=${unit}&appid=${appId}`, { mode: 'cors' });
+  const json = await response.json();
```
Comment (uryelah) [inline src/weather_api.js]: - [x] Add error handling to your async calls, that can be done with [try/catch blocks](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch).

File: src/dom.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+const displayErrorMessage = (cityName) => {
+  const resultContainer = document.getElementById('result');
+  resultContainer.innerHTML = '';
+  const city = document.createElement('p');
+  city.classList.add('error');
+  resultContainer.appendChild(city);
+  city.textContent = `Sorry, data for "${cityName}" is unavailable`;
+};
+
+const displayResult = (result, tempUnit) => {
```
Comment (uryelah) [inline src/dom.js:14]: - [ ] [OPTIONAL] Consider changing also the app background according to the weather result, that and working more on your UI will make this project a good candidate for your portfolio.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Simon Misori** , I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with this project, you worked really well with webpack and separating your DOM and logic. There are still a couple of fixes to work on before getting it approved so hang in there.

Don't get discouraged though, you've done most of the job, just go on for a little more.

Read the points I mention below, work on the fixes, and submit for a new review.

Keep on coding and Rock on! 🎸 👨‍🎤 

# Require Changes ♻️ 

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by christianotieno — APPROVED** [review]
## Project Approved 🍾 
Hello @Misori-simon 👋🏾 
Congratulations. The project is approved. 👏🏾 
All the very best in your next tasks 🥇 
### Happy coding 💻 
#### Christian Otieno (CR)

**Pull request merged**